__init__.py

```diff
@@ -98,137 +77,90 @@ def add_routine(self, message):
+        if self.routines[rid].get("active"):
+            if rid in self.scheduled_routines:
+                self.cancel_scheduled_event(rid)
+            self.setup_routine_events(self.routines[rid])
 
-        for i, r in enumerate(self.routines_model["routines"]):
-            if r["id"] == routine["id"]:
-                self.routines_model["routines"][i] = routine
-                break
-        self.routines_model.store()
-
-        if routine in self.active_routines:
-            self.active_routines.remove(routine)
-            self.active_routines.append(routine)
-            self.active_routines_storage["routines"] = self.active_routines
-            self.active_routines_storage.store()
-            
-            if routine in self.scheduled_routines:
-                self.cancel_scheduled_event(routine["id"])
-                self.setup_routine_events(routine)
-        
-        if routine in self.inactive_routines:
-            self.inactive_routines.remove(routine)
-            self.inactive_routines.append(routine)
-            self.inactive_routines_storage["routines"] = self.inactive_routines
-            self.inactive_routines_storage.store()
-    
     def remove_routine_gui(self, message):
         routine_id = message.data["routine_id"]
         self.remove_routine(routine_id)
 
     def remove_routine(self, routine_id):
-        routine = self.get_routine_by_id(routine_id)
-        if routine in self.active_routines:
-            self.active_routines.remove(routine)
-            self.active_routines_storage["routines"] = self.active_routines
-            self.active_routines_storage.store()
-
-            self.cancel_scheduled_event(routine_id)
-            if routine_id in self.scheduled_routines:
-                self.scheduled_routines.remove(routine_id)
-
-        if routine in self.inactive_routines:
-            self.inactive_routines.remove(routine)
-            self.inactive_routines_storage["routines"] = self.inactive_routines
-            self.inactive_routines_storage.store()
-    
-        self.routines_model["routines"].remove(routine)
-        self.routines_model.store()
-        
-        self.gui["routines_model"] = self.routines_model["routines"]
-        self.gui["active_routines"] = self.active_routines_storage
-        self.gui["inactive_routines"] = self.inactive_routines_storage
+        if routine_id in self.routines:
+            routine = self.routines[routine_id]
+            if routine.get("active"):
+                self.cancel_scheduled_event(routine_id)
+                if routine_id in self.scheduled_routines:
+                    self.scheduled_routines.remove(routine_id)
+
+            self.routines.pop(routine_id)
+            self._update_gui()
+
+    def _update_gui(self):
+        self.gui["routines_model"] = list(self.routines.values())
+        self.gui["active_routines"] = {"routines": self.active_routines}
+        self.gui["inactive_routines"] = {"routines": self.inactive_routines}
```

Comment:
which one is correct? in method below you do both
```python
self.gui["active_routines"] = self.active_routines_storage["routines"]
self.gui["active_routines"] = self.active_routines_storage
```

Reply:
self.gui["active_routines"] = self.active_routines_storage